Add auto-enrollment subsection in client (#2433)
wazuh/wazuh-documentation: source/user-manual/reference/ossec-conf/client.rst

      <time-reconnect>120</time-reconnect>
      <auto_restart>yes</auto_restart>
    </client>

enrollment
^^^^^^^^^^

.. versionadded:: 4.0

Configures the connection parameters for the agent enrollment.

Options
-------

- `enabled`_
- `manager_address`_
- :ref:`port <enrollment_port>`
- `agent_name`_
- `groups`_
- `agent_address`_
- `ssl_cipher`_
- `server_ca_path`_
- `agent_certificate_path`_
- `agent_key_path`_
- `authorization_pass`_
- `auto_method`_
- `delay_after_enrollment`_
- `use_source_ip`_

enabled
^^^^^^^

> Enables/Disables agent enrollment.

Reviewer: ```suggestion
Enables/disables agent enrollment.
```

| **Allowed values** | string -  Should be valid IP/Hostname |
+--------------------+---------------------------------------+

.. _enrollment_port:

port
^^^^

Specifies the port on the manager to send enrollment request.  This must match the associated listening port configured on the Wazuh manager.

+--------------------+---------------------------------------------+
| **Default value**  | 1515                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | Any port number from 0 to 65535 is allowed. |
+--------------------+---------------------------------------------+

agent_name
^^^^^^^^^^

Agent name that will be used for enrollment. Only alphanumeric characters, "-", "_" or "." are allowed, and the minimum length is two characters.

+--------------------+---------------------------------------------+
| **Default value**  | Hostname of the machine.                    |
+--------------------+---------------------------------------------+
| **Allowed values** | string - Registration name for the agent.   |
+--------------------+---------------------------------------------+

groups
^^^^^^

> Groups name to which the agent belong.

Reviewer: ```suggestion
Groups name to which the agent belongs.
```

+--------------------+---------------------------------------------+
| **Default value**  | NULL                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | string - Name of one or many valid groups.  |
+--------------------+---------------------------------------------+

agent_address
^^^^^^^^^^^^^

> Force IP address from agent. If this is not set manager will extract source IP from enrollment messsage.

Reviewer: ```suggestion
Force IP address from the agent. If this is not set manager will extract source IP from enrollment message.
```

| **Default value**  | src                                         |
+--------------------+---------------------------------------------+
| **Allowed values** | string - Valid IP.                          |
+--------------------+---------------------------------------------+

ssl_cipher
^^^^^^^^^^

Override SSL used ciphers.

+--------------------+---------------------------------------------+
| **Default value**  | DEFAULT_CIPHERS                             |
+--------------------+---------------------------------------------+
| **Allowed values** | valid ssl ciphers.                          |
+--------------------+---------------------------------------------+

server_ca_path
^^^^^^^^^^^^^^

Used for manager verification. If no CA certificate is set server will not be verified.

+--------------------+---------------------------------------------+
| **Default value**  | NULL                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | Path to a valid CA certificate.             |
+--------------------+---------------------------------------------+

agent_certificate_path
^^^^^^^^^^^^^^^^^^^^^^

> Required when agent verification is enabled in manager.

Reviewer: ```suggestion
Required when agent verification is enabled in the manager.
```

+--------------------+---------------------------------------------+
| **Default value**  | NULL                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | Path to a valid agent certificate file.     |
+--------------------+---------------------------------------------+

agent_key_path
^^^^^^^^^^^^^^

> Required when agent verification is enabled in manager.

Reviewer: ```suggestion
Required when agent verification is enabled in the manager.
```

+--------------------+---------------------------------------------+
| **Default value**  | NULL                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | Path to a valid agent key file.             |
+--------------------+---------------------------------------------+

authorization_pass
^^^^^^^^^^^^^^^^^^

Required when enrollment is using password verification. The password must match that of the manager.

+--------------------+---------------------------------------------+
| **Default value**  | NULL                                        |
+--------------------+---------------------------------------------+
| **Allowed values** | Password string.                            |
+--------------------+---------------------------------------------+

auto_method
^^^^^^^^^^^

> Auto negotiate the most secure common SSL/TLS method with the manager, use "yes" for auto negotiate or "no" for TLS v1.2 only.

Reviewer: ```suggestion
Auto negotiates the most secure common SSL/TLS method with the manager, use "yes" for auto negotiate or "no" for TLS v1.2 only.
```

+--------------------+---------------------------------------------+
| **Default value**  | no                                          |
+--------------------+---------------------------------------------+
| **Allowed values** | yes or no                                   |
+--------------------+---------------------------------------------+

delay_after_enrollment
^^^^^^^^^^^^^^^^^^^^^^

> Time agentd should wait after a successfull registration.

Reviewer: ```suggestion
Time that agentd should wait after a successful registration.
```
